### ISSUE TITLE
[Feature Request] Improve event slot for VCalendar

### ISSUE THREAD

**Issue by ClickerMonkey** [vuetifyjs/vuetify#11133]
### Problem to solve
You have access to the original event, but the parsed event is essential for common calendar functionality. It would also be nice to have access to functions which are used to display the event name and potentially a summary of the time.

### Proposed solution
Add to slot scope
- parsedEvent
- getEventName
- formatTime
- singleline

**Comment by ClickerMonkey** [comment]
Playground for this feature:

<details>

```vue

<template>
  <v-container>
    <div>
      <v-sheet
        tile
        height="54"
        color="grey lighten-3"
        class="d-flex"
      >
        <v-btn
          icon
          class="ma-2"
          @click="$refs.calendar.prev()"
        >
          <v-icon>mdi-chevron-left</v-icon>
        </v-btn>
        <v-spacer></v-spacer>
        <v-btn
          icon
          class="ma-2"
          @click="$refs.calendar.next()"
        >
          <v-icon>mdi-chevron-right</v-icon>
        </v-btn>
      </v-sheet>
      <v-sheet height="600">
        <v-calendar
          ref="calendar"
          v-model="value"
          event-name="name"
          :type="type"
          :events="events"
          :event-color="getEventColor"
          @change="getEvents"
          @click:event="changeName"
        >
          <template #event="{ event, eventSummary, timed }">
            <div class="pl-1">
              <v-icon dark x-small>{{ event.icon }}</v-icon>
              <span v-html="eventSummary()"></span>
            </div>
            <div v-if="timed"
              class="v-event-drag-bottom"
            ></div>
          </template>
        </v-calendar>
      </v-sheet>
    </div>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      type: 'week',
      mode: 'stack',
      value: '',
      events: [],
      colors: ['blue', 'indigo', 'deep-purple', 'cyan', 'green', 'orange', 'grey darken-1'],
      names: ['Meeting', 'Holiday', 'PTO', 'Travel', 'Event', 'Birthday', 'Conference', 'Party'],
      icons: ['mdi-fire', 'mdi-pine-tree', 'mdi-star', 'mdi-car', 'mdi-calendar', 'mdi-cake', 'mdi-webcam', 'mdi-party-popper'],
    }),
    methods: {
      getEvents ({ start, end }) {
        const events = []

        const min = new Date(`${start.date}T00:00:00`)
        const max = new Date(`${end.date}T23:59:59`)
        const days = (max.getTime() - min.getTime()) / 86400000
        const eventCount = this.rnd(days, days + 20)

        for (let i = 0; i < eventCount; i++) {
          const allDay = this.rnd(0, 3) === 0
          const firstTimestamp = this.rnd(min.getTime(), max.getTime())
          const first = new Date(firstTimestamp - (firstTimestamp % 900000))
          const secondTimestamp = this.rnd(2, allDay ? 288 : 8) * 900000
          const second = new Date(first.getTime() + secondTimestamp)
          const type = this.rnd(0, this.names.length - 1)

          events.push({
            name: this.names[type],
            icon: this.icons[type],
            start: first,
            end: second,
            color: this.colors[this.rnd(0, this.colors.length - 1)],
            timed: !allDay,
          })
        }

        this.events = events
      },
      getEventColor (event) {
        return event.color
      },
      changeName ({ event }) {
        event.name += '*'
      },
      rnd (a, b) {
        return Math.floor((b - a + 1) * Math.random()) + a
      },
    },
  }
</script>

<style>
.v-event-drag-bottom {
  position: absolute; 
  left: 0; 
  right: 0;
  bottom: 4px; 
  height: 4px; 
  cursor: ns-resize;
}
.v-event-drag-bottom::after {
  display: none;
  position: absolute; 
  left: 50%; 
  height: 4px; 
  border-top: 1px solid white; 
  border-bottom: 1px solid white; 
  width: 20px; 
  margin-left: -10px;
  opacity: 0.8;
  content: '';
}
.v-event-timed:hover .v-event-drag-bottom::after {
  display: block;
}
</style>
```

</details>

**Comment by johnleider** [comment]
resolved in #11198

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).
